### ISSUE TITLE
PINN can't backup Raspberry Pi OS

### ISSUE THREAD

**Issue by realOgenki** [procount/pinn#398]
When I tried to backup my OS with PINN a splash windows show me this message:

![Screenshot_20200703_130419](https://user-images.githubusercontent.com/33685025/86463708-4c089e80-bd2e-11ea-947e-ce2bc06f519c.jpg)

**Comment by procount** [comment]
OOPS, small problem since they renamed their OS.
Please reboot PINN **TWICE** whilst connected to the internet, and then it should be ok.

**Comment by lurch** [comment]
LOL, I didn't realise PINN suffered from the "turn it off and on again" meme too :rofl:

**Comment by procount** [comment]
:smiley: You can't get an easier support solution than that, though, can you!

**Comment by realOgenki** [comment]
> OOPS, small problem since they renamed their OS.
> Please reboot PINN **TWICE** whilst connected to the internet, and then it should be ok.

"Microsoft solution" works, but I found another issue:

![Screenshot_20200703_141218](https://user-images.githubusercontent.com/33685025/86468330-8296e700-bd37-11ea-84ba-0d466165971a.jpg)

Where is log file and how I can solve it?

**Comment by procount** [comment]
see https://github.com/procount/pinn/wiki/Troubleshooting for info.

**Comment by realOgenki** [comment]
> see https://github.com/procount/pinn/wiki/Troubleshooting for info.

Thanks for your advice and help but I connect to PINN through VNC and I don't know the way to see and use the recovery shell.
Any idea?
Regards

**Comment by procount** [comment]
Go to maintenance, select PINN, edit config, Edit Options and set SSH and reboot.
Then you can log in with ssh (root/raspberry) and do the same things as if you were in the recovery shell.
But the reboot will use your logs, so you'll have to try backing up again to repeat the error.

**Comment by realOgenki** [comment]
> Go to maintenance, select PINN, edit config, Edit Options and set SSH and reboot.
> Then you can log in with ssh (root/raspberry) and do the same things as if you were in the recovery shell.
> But the reboot will use your logs, so you'll have to try backing up again to repeat the error.

Great! I had not seen ssh option on edit option before. I have did it with a direct connection and I show you the last lines:

```
"sh -c "cd /tmp/src; tar -c . | pigz > /tmp/media/sda1/os/Raspberry_Pi_OS_(32-bit)#20200703-160533/boot.tar.gz"" 
sh: syntax error: unexpected "("
"Error writing Raspberry Pi OS (32-bit): Disk full?" 
```

My USB is not full, maybe it could be a issue with the parethesis...

**Comment by procount** [comment]
Yes, looks like an issue with parentheses. I thought I had fixed that, but looks like I missed one.
Bear with me, it might take a little while for me to fix it.

**Comment by realOgenki** [comment]
> Yes, looks like an issue with parentheses. I thought I had fixed that, but looks like I missed one.
> Bear with me, it might take a little while for me to fix it.

Don't worry, you are the developer and I have a lot of patience ;-)
Thank you.

**Comment by realOgenki** [comment]
Any news about this issue? I hope you didn't forget it.. ;-)

**Comment by procount** [comment]
Sorry, I'm struggling with resources at the moment (time & laptop). 
I hope to have it done soon.

**Comment by realOgenki** [comment]
Hi, procount.
Recently I tried again to backup my Raspberry Pi OS because I can't an easy way to resize my SD partitions, but PINN show me the same issue. If I can edit OS name when I backup the system, I could delete the parenthesis... Some progress?
Regards.

**Comment by procount** [comment]
I'm really sorry I've not been able to look at this for you yet. I have some really pressing work to do at the moment and some issues to resolve, but I will get back to this as soon as those are out of the way.

**Comment by procount** [comment]
Fixed in v3.3.4.3 Just released.

**Comment by realOgenki** [comment]
> Fixed in v3.3.4.3 Just released.

Checked! Thank you very much, procount.
I have a question: could I restore the backup on a partition of different size than original?

**Comment by procount** [comment]
Yes, provided it is big enough to store all the content. Plus you need some additional working spare disk capacity (for temp files, apt upgrade etc.) otherwise you may find you haven't got enough disk space to boot it. The boot partition will be the same size, but the main root partition can vary

**Comment by realOgenki** [comment]
Thank you again, procount.
Could you tell me something more about this additional working spare disk capacity?

**Comment by procount** [comment]
Not really. It depends on your OS and what you will do with it.
All I'm saying is that if you use a general purpose OS like Raspbian /RasPiOS and your backup is 8GB in size, don't put it on a partition that is only 8.1GB, otherwise you will soon run out of disk space and it may stop working.

**Comment by realOgenki** [comment]
I understand, procount. Thank you for your help and comments. My best wishes for this fantastic project.

**Comment by Retro-Devils** [comment]
So im having a similar issue on newest PINN. The backup button wont even let me select it at all. Very confusing lol. 

Hoping to get some  help here. Tried reboot twice lol

**Comment by cRoCx** [comment]
> > OOPS, small problem since they renamed their OS.
> > Please reboot PINN **TWICE** whilst connected to the internet, and then it should be ok.
> 
> "Microsoft solution" works, but I found another issue:
> 
> ![Screenshot_20200703_141218](https://user-images.githubusercontent.com/33685025/86468330-8296e700-bd37-11ea-84ba-0d466165971a.jpg)
> 
> Where is log file and how I can solve it?

Sorry to open this up again, but I have just encountered the same error message (see screenshot) and after some debugging I think I found the cause:
1. Debug log says it can't write more data -> Looked up free space on backup device -> Still >100 GB available for a backup of only 17 GB. But ... it could only write 4 GB -> This is the file size limit on FAT32.
2. PINN does recognise other file systems which allow for larger files and recognises them i.e. at the start screen, but it just hides them when you go to the other menu where you do backups. The list emties out there. Happened to me with ext4 and exFAT.
Conclusion: You can only backup to FAT32 with the 4GB file size limitation. All other file systems seem unavailable to the backup functionality.

**Comment by procount** [comment]
If the drive is hidden on the Archive and Maintenance menus, I suspect you forgot to add the 'os' folder to the drive first. Otherwise it will not be recognized as a backup device. 
I just tested backing up a 13GB OS to an ext4 drive successfully.

**Comment by cRoCx** [comment]
Oh yes, I think I forgot about that. My bad. :-( After the FAT32 backup finished with errors (>4GB), I think I started fiddling around, trying to find a solution for this, formatting and reformatting SD cards with other file system types, totally forgetting about about this `/os` folder, which must be added manually each time. ... and I also overlooked the sentence with this hint in the readme ... after growing more and more desperation.

Could you add the appropriate formatting to the word "/os" in the backup section, like this.

Before: The backup function allows an installed OS to be backed up to a USB drive. The USB drive must first be formatted as FAT32, ext4 or ntfs and have a /os folder on it to store the backups in, otherwise it will not be recognised.

After: The backup function allows an installed OS to be backed up to a USB drive. The USB drive must first be formatted as FAT32, ext4 or ntfs and have a `/os` folder on it to store the backups in, otherwise it will not be recognised.

In other sections of the readme the `/os` stands out like this, but not in the _backup_ section. I could also create a pull request for this, if you want.

Sorry that I wasted a little of your time, since it was clearly my mistake somehow.

**Comment by procount** [comment]
No worries, I updated the backup section.
